app/models/name/taxonomy.rb

```diff
@@ -452,10 +452,6 @@ def has_notes?
     notes&.match(/\S/)
   end
 
-  def text_before_rank
-    text_name.split(" #{rank.to_s.downcase}").first
-  end
-
```

Comment:
This is dead code (and apparently was never used).And it's buggy. 
For the (embarrasing) details, see https://www.pivotaltracker.com/n/projects/224629/stories/179474221/